nifi-nar-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/processors/slack/PutSlack.java

```diff
@@ -0,0 +1,245 @@
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.stream.io.DataOutputStream;
+
+import javax.json.Json;
+import javax.json.JsonObject;
+import javax.json.JsonObjectBuilder;
+import javax.json.JsonWriter;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@Tags({"put", "slack", "notify"})
+@CapabilityDescription("Sends a message to your team on slack.com")
+@InputRequirement(InputRequirement.Requirement.INPUT_REQUIRED)
+public class PutSlack extends AbstractProcessor {
+
+    public static final PropertyDescriptor WEBHOOK_URL = new PropertyDescriptor
+            .Builder()
+            .name("webhook-url")
+            .displayName("Webhook URL")
+            .description("The POST URL provided by Slack to send messages into a channel.")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .addValidator(StandardValidators.URL_VALIDATOR)
+            .sensitive(true)
+            .build();
+
+    public static final PropertyDescriptor WEBHOOK_TEXT = new PropertyDescriptor
+            .Builder()
+            .name("webhook-text")
+            .displayName("Webhook Text")
+            .description("The text sent in the webhook message")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor CHANNEL = new PropertyDescriptor
+            .Builder()
+            .name("channel")
+            .displayName("Channel")
+            .description("A public channel using #channel or direct message using @username. If not specified, " +
+                    "the default webhook channel as specified in Slack's Incoming Webhooks web interface is used.")
+            .required(false)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor USERNAME = new PropertyDescriptor
+            .Builder()
+            .name("username")
+            .displayName("Username")
+            .description("The displayed Slack username")
+            .required(false)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor ICON_URL = new PropertyDescriptor
+            .Builder()
+            .name("icon-url")
+            .displayName("Icon URL")
+            .description("Icon URL to be used for the message")
```

Comment:
Oh boy. Well if it's an inconsistent interface on their part, then there's not much we can do at this point. Everything else looks good, I will merge to 0.x and master. Thanks again for this contribution!